MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs

```diff
@@ -7559,5 +7562,39 @@ private void hdluid()
         ///     Signature: struct uidxrf;
         /// </summary>
         public ReadOnlySpan<byte> uidxrf => Module.Memory.GetVariablePointer("UIDXRF").Data;
+
+        /// <summary>
+        ///     Allocate new space for a string
+        ///
+        ///     Signature: char *alcdup(char *stg);
+        /// </summary>
+        private void alcdup()
+        {
+            var sourceStringPointer = GetParameterPointer(0);
+            var inputBuffer = Module.Memory.GetString(sourceStringPointer);
+            var destinationAllocatedPointer = Module.Memory.AllocateVariable(null, (ushort)inputBuffer.Length);
+
+
+            if (inputBuffer[0] == 0x0 || sourceStringPointer == IntPtr16.Empty)
```

Comment:
Can probably remove `inputBuffer[0] == 0x0`, since the `SetArray` call can handle a single byte array